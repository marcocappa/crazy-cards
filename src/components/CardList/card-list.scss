.crazy-card-cards-list {
    margin: 0 -1rem;
    padding: 0;
    display: flex;
    list-style: none;
    flex-wrap: wrap;

    &__item {
        margin: 0 1rem 1rem;
        border-radius: 1rem;
        border: 0.0625rem solid #e2e5e8;
        background-color: white;
        box-sizing: border-box;
        padding: 1rem;
        min-width: 20.875rem;
        width: auto;
    }
}
