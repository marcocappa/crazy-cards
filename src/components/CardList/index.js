import React from 'react';
import './card-list.scss';
import CardItem from '../CardItem';

const CardList = ({ items }) => (
  <ul className="crazy-card-cards-list">
    {items.map((item) => (
      <CardItem
        {...item}
        key={item.id}
        className="crazy-card-cards-list__item"
      />
    ))}
  </ul>
);

export default CardList;
