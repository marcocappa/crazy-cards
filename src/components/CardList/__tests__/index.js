import React from 'react';
import CardList from '../index';
import { shallow } from 'enzyme';
import { beforeEach, expect } from '@jest/globals';

describe('<CardList />', () => {
  const mockProps = {
    items: [
      {
        id: 'card-0',
        name: 'Student Life',
        apr: 18.9,
        btod: 0,
        pod: 6,
        creditAvailable: 1200,
      },
      {
        id: 'card-1',
        name: 'Anywhere Card',
        apr: 33.9,
        btod: 0,
        pod: 0,
        creditAvailable: 300,
      },
    ],
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CardList {...mockProps} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should find ul children to be equal to prop items length', () => {
    expect(wrapper.find('ul').children().length).toEqual(
      mockProps.items.length
    );
  });
});
