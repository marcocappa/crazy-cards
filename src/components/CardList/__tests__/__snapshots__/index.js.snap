// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> should match snapshot 1`] = `
<ul
  className="crazy-card-cards-list"
>
  <CardItem
    apr={18.9}
    btod={0}
    className="crazy-card-cards-list__item"
    creditAvailable={1200}
    id="card-0"
    key="card-0"
    name="Student Life"
    pod={6}
  />
  <CardItem
    apr={33.9}
    btod={0}
    className="crazy-card-cards-list__item"
    creditAvailable={300}
    id="card-1"
    key="card-1"
    name="Anywhere Card"
    pod={0}
  />
</ul>
`;
